core/src/main/scala/io/qbeast/core/model/RevisionClasses.scala

```diff
@@ -152,7 +152,15 @@ final case class Revision(
    *   true if the revision indexes all and only the provided columns.
    */
   def matchColumns(columnsToIndex: Seq[String]): Boolean = {
-    columnsToIndex.toSet == columnTransformers.map(_.columnName).toSet
+    columnsToIndex.size == columnTransformers.size && columnsToIndex
+      .zip(columnTransformers)
+      .forall { case (columnToIndex, t) =>
+        if (columnToIndex.contains(":")) {
+          columnToIndex == t.spec
```

Comment:
Shouldn't we compare the name in the `columnToIndex` with the name in `t` before?

Reply:
When `columnToIndex` contains `":"` it is a spec in the form of `"col_1:linear"`. In other words, the user here is specifying the transformation it wants to do with `col_1`.

Reply:
oohhh, I see...

I think it's safer to use Regex to enforce the format of columnName:transformationType and compare each other separately.